src/simulations/set_generator.py

```diff
@@ -51,6 +52,48 @@ def __iter__(self):
     for _ in range(self.num_sets):
       yield self.set_size
 
+def _choice_fast(n, m, random_state):
+    """O(m) space-optimal algorithm for generating m random indices for list 
+    of size n without replacement
+    
+    Args:
```

Comment:
See the style guide on comments for functions and methods ...

Reply:
Resolved (I think!) in c5bdec7